lib/speech/index.js

```diff
@@ -0,0 +1,903 @@
+ *     Default: `false`.
+ * @param {string[]=} asyncRecognizeRequest.config.speechContext - See
+ *     [`RecognitionConfig.speechContext`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `[]`.
+ * @param {object} asyncRecognizeRequest.audio - A `RecognitionAudio` object.
+ *     See [`RecognitionAudio`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionAudio).
+ * @param {buffer} asyncRecognizeRequest.audio.content - The audio data bytes
+ *     encoded as specified in `RecognitionConfig`.
+ * @param {string} asyncRecognizeRequest.audio.uri - URI that points to a file
+ *     that contains audio data bytes as specified in `RecognitionConfig`.
+ *     Currently, only Google Cloud Storage URIs are supported, which must be
+ *     specified in the following format: `gs://bucket_name/object_name`.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:speech/operation} callback.operation - `Operation` object. See
+ *     [`Operation`](https://cloud.google.com/speech/reference/rpc/google.longrunning#google.longrunning.Operation).
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * //-
+ * // <h3>Send raw audio</h3>
+ * //-
+ * var fs = require('fs');
+ *
+ * fs.readFile('./system-test/data/bridge.raw', function (err, rawAudio) {
+ *   // See the link in the parameters for `AsyncRecognizeRequest`.
+ *   var asyncRecognizeRequest = {
+ *     config: {
+ *       encoding: 'LINEAR16',
+ *       sampleRate: 16000
+ *     },
+ *     audio: {
+ *       content: rawAudio
+ *     }
+ *   };
+ *
+ *   speech.createRecognizeJob(
+ *     asyncRecognizeRequest,
+ *     function(err, operation, apiResponse) {
+ *       console.log(operation);
+ *     }
+ *   );
+ * });
+ *
+ * //-
+ * // <h3>Specify URI to audio file</h3>
+ * //-
+ * // See the link in the parameters for `AsyncRecognizeRequest`.
+ * var asyncRecognizeRequest = {
+ *   config: {
+ *     encoding: 'LINEAR16',
+ *     sampleRate: 16000
+ *   },
+ *   audio: {
+ *     uri: 'gs://your-bucket-name/bridge.raw'
+ *   }
+ * };
+ *
+ * speech.createRecognizeJob(
+ *   asyncRecognizeRequest,
+ *   function(err, operation, apiResponse) {
+ *     console.log(operation);
+ *   }
+ * );
+ */
+Speech.prototype.createRecognizeJob = function(reqOpts, callback) {
+  var self = this;
+  var protoOpts = {
+    service: 'Speech',
+    method: 'asyncRecognize'
+  };
+
+  this.request(protoOpts, reqOpts, function(err, apiResponse) {
+    if (err) {
+      callback(err, null, apiResponse);
+      return;
+    }
+    var decoder = self.protos.Speech.AsyncRecognizeResponse.decode;
+    callback(null, self.operation(apiResponse, decoder), apiResponse);
+  });
+};
+
+/**
+ * Asynchronously detect speech from an audio file.
```

Comment:
Could we add some verbiage to the `async` requests stating when you might want to use them instead of their synchronous counterpart?